File: IQ/Classes/clsClickThru.vb
Option Strict On
Option Explicit On

Imports System.Data.SqlClient

''' <summary>
''' 
''' </summary>
''' <remarks></remarks>

Public Class clsClickThru
    Implements i_Editable

    Property ID As Integer
    Property Account As clsAccount
    Property Advert As clsAdvert
    Property TimeStamp As Date
    Private conString As String = ConfigurationManager.ConnectionStrings("DBConnectString").ConnectionString

    Public Sub New()


    End Sub
    Public Sub New(account As clsAccount, advert As clsAdvert, timestamp As Date)

        Dim con As SqlConnection = New SqlConnection(conString)
        con.Open()
        Dim command As SqlCommand = New SqlCommand()
        command.CommandText = "AddClickThru"
        command.CommandType = CommandType.StoredProcedure
        command.Connection = con
        Dim paramUserID As New SqlParameter("@accountid", SqlDbType.Int)
        paramUserID.Value = account.ID
        Dim paramAdvertID As New SqlParameter("@advertid", SqlDbType.Int)
        paramAdvertID.Value = advert.ID
        Dim paramTimeStamp As New SqlParameter("@timestamp", SqlDbType.DateTime)
        paramTimeStamp.Value = timestamp

        Dim paramReturn As New SqlParameter("@ret", SqlDbType.Int)
        paramReturn.Direction = ParameterDirection.ReturnValue

        command.Parameters.Add(paramUserID)
        command.Parameters.Add(paramAdvertID)
        command.Parameters.Add(paramTimeStamp)
        command.Parameters.Add(paramReturn)

        command.ExecuteNonQuery()

        con.Close()

        Me.ID = Convert.ToInt32(paramReturn.Value)
        Me.Account = account
        Me.Advert = advert
        Me.TimeStamp = timestamp

        Me.Advert.ClickThrus.Add(Me.ID, Me)
        Me.Account.ClickThrus.Add(Me.ID, Me)

    End Sub
    Public Sub New(ID As Integer, user As clsUser, advert As clsAdvert, timestamp As Date)
        Me.ID = ID
        Me.Account = Account
        Me.Advert = advert
        Me.TimeStamp = timestamp

        Me.Advert.ClickThrus.Add(Me.ID, Me)
        Me.Account.ClickThrus.Add(Me.ID, Me)

    End Sub
    Public Sub delete(ByRef errorMessages As List(Of String)) Implements i_Editable.delete
        Me.Advert.ClickThrus.Remove(Me.ID)
        Me.Account.ClickThrus.Remove(Me.ID)
    End Sub

    Public Function displayName(Language As clsLanguage) As String Implements i_Editable.displayName

    End Function

    Public Function Insert(ByRef errorMessages As List(Of String)) As Object Implements i_Editable.Insert
        Return New clsClickThru(Me.Account, Me.Advert, Me.TimeStamp)
    End Function

    Public Sub update(ByRef errorMessages As List(Of String)) Implements i_Editable.update
        If Me.ID > 0 Then

            Dim con As SqlConnection = New SqlConnection(conString)
            con.Open()
            Dim command As SqlCommand = New SqlCommand()
            command.CommandText = "UpdateClickThru"
            command.CommandType = CommandType.StoredProcedure
            command.Connection = con
            Dim paramID As New SqlParameter("@ID", SqlDbType.Int)
            paramID.Value = Me.ID
            Dim paramUserID As New SqlParameter("@accountid", SqlDbType.Int)
            paramUserID.Value = Account.ID
            Dim paramAdvertID As New SqlParameter("@adverid", SqlDbType.Int)
            paramAdvertID.Value = Advert.ID
            Dim paramTimeStamp As New SqlParameter("@url", SqlDbType.DateTime)
            paramTimeStamp.Value = TimeStamp

            Dim paramReturn As New SqlParameter("@ret", SqlDbType.Int)
            paramReturn.Direction = ParameterDirection.ReturnValue

            command.Parameters.Add(paramUserID)
            command.Parameters.Add(paramAdvertID)
            command.Parameters.Add(paramTimeStamp)
            command.Parameters.Add(paramReturn)

            command.ExecuteNonQuery()

            con.Close()
        End If
    End Sub
End Class

File: IQ/Classes/clsClickThru.cs
using System.Data.SqlClient;



/// <summary>
///
/// </summary>
/// <remarks></remarks>

public class clsClickThru : i_Editable
{

    public int ID { get; set; }
    public clsAccount Account { get; set; }
    public clsAdvert Advert { get; set; }
    public DateTime TimeStamp { get; set; }
    private string conString; // VBConversions Note: Initial value cannot be assigned here since it is non-static.  Assignment has been moved to the class constructors.

    public clsClickThru()
    {
        // VBConversions Note: Non-static class variable initialization is below.  Class variables cannot be initially assigned non-static values in C#.
        conString = System.Convert.ToString(ConfigurationManager.ConnectionStrings("DBConnectString").ConnectionString);



    }
    public clsClickThru(clsAccount account, clsAdvert advert, DateTime timestamp)
    {
        // VBConversions Note: Non-static class variable initialization is below.  Class variables cannot be initially assigned non-static values in C#.
        conString = System.Convert.ToString(ConfigurationManager.ConnectionStrings("DBConnectString").ConnectionString);


        SqlConnection con = new SqlConnection(conString);
        con.Open();
        SqlCommand command = new SqlCommand();
        command.CommandText = "AddClickThru";
        command.CommandType = CommandType.StoredProcedure;
        command.Connection = con;
        SqlParameter paramUserID = new SqlParameter("@accountid", SqlDbType.Int);
        paramUserID.Value = Account.ID;
        SqlParameter paramAdvertID = new SqlParameter("@advertid", SqlDbType.Int);
        paramAdvertID.Value = Advert.ID;
        SqlParameter paramTimeStamp = new SqlParameter("@timestamp", SqlDbType.DateTime);
        paramTimeStamp.Value = timestamp;

        SqlParameter paramReturn = new SqlParameter("@ret", SqlDbType.Int);
        paramReturn.Direction = ParameterDirection.ReturnValue;

        command.Parameters.Add(paramUserID);
        command.Parameters.Add(paramAdvertID);
        command.Parameters.Add(paramTimeStamp);
        command.Parameters.Add(paramReturn);

        command.ExecuteNonQuery();

        con.Close();

        this.ID = System.Convert.ToInt32(Convert.ToInt32(paramReturn.Value));
        this.Account = account;
        this.Advert = advert;
        this.TimeStamp = timestamp;

        this.Advert.ClickThrus.Add(this.ID, this);
        this.Account.ClickThrus.Add(this.ID, this);

    }
    public clsClickThru(int ID, clsUser user, clsAdvert advert, DateTime timestamp)
    {
        // VBConversions Note: Non-static class variable initialization is below.  Class variables cannot be initially assigned non-static values in C#.
        conString = System.Convert.ToString(ConfigurationManager.ConnectionStrings("DBConnectString").ConnectionString);

        this.ID = ID;
        this.Account = Account;
        this.Advert = advert;
        this.TimeStamp = timestamp;

        this.Advert.ClickThrus.Add(this.ID, this);
        this.Account.ClickThrus.Add(this.ID, this);

    }
    public void delete(ref List<string> errorMessages)
    {
        this.Advert.ClickThrus.Remove(this.ID);
        this.Account.ClickThrus.Remove(this.ID);
    }

    public string displayName(clsLanguage Language)
    {

    }

    public dynamic Insert(ref List<string> errorMessages)
    {
        return new clsClickThru(this.Account, this.Advert, this.TimeStamp);
    }

    public void update(ref List<string> errorMessages)
    {
        if (this.ID > 0)
        {

            SqlConnection con = new SqlConnection(conString);
            con.Open();
            SqlCommand command = new SqlCommand();
            command.CommandText = "UpdateClickThru";
            command.CommandType = CommandType.StoredProcedure;
            command.Connection = con;
            SqlParameter paramID = new SqlParameter("@ID", SqlDbType.Int);
            paramID.Value = this.ID;
            SqlParameter paramUserID = new SqlParameter("@accountid", SqlDbType.Int);
            paramUserID.Value = Account.ID;
            SqlParameter paramAdvertID = new SqlParameter("@adverid", SqlDbType.Int);
            paramAdvertID.Value = Advert.ID;
            SqlParameter paramTimeStamp = new SqlParameter("@url", SqlDbType.DateTime);
            paramTimeStamp.Value = TimeStamp;

            SqlParameter paramReturn = new SqlParameter("@ret", SqlDbType.Int);
            paramReturn.Direction = ParameterDirection.ReturnValue;

            command.Parameters.Add(paramUserID);
            command.Parameters.Add(paramAdvertID);
            command.Parameters.Add(paramTimeStamp);
            command.Parameters.Add(paramReturn);

            command.ExecuteNonQuery();

            con.Close();
        }
    }
}